### ISSUE TITLE
Use mod dependency information from Workshop

### ISSUE THREAD

**Issue by robojumper** [X2CommunityCore/xcom2-launcher#56]
The Steam API supports mod dependencies. We should investigate how they are listed in API responses and add a a way for dependencies to be listed for mods published to Steam.

**Comment by RevZero** [comment]
Did anyone who is familiar with the steam API already check if this is possible?

**Comment by Xymanek** [comment]
https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/pull/525 allows mods to specify their requirements in config - we should make use of that. Examples:

* https://github.com/WOTCStrategyOverhaul/CovertInfiltration/blob/f12cf8583fa75d0fc019e3e17884b9407e8a4c73/CovertInfiltration/Config/XComGame.ini#L33-L53
* https://github.com/Musashi1584/RPGO/blob/705fa87958932e62667bfee6f6ee244137d4a7d2/XCOM2RPGOverhaul/Config/XComGame.ini#L31-L49

Also, the steam provides an API for fetching the workshop requirements: https://partner.steamgames.com/doc/api/ISteamUGC#GetAppDependencies

**Comment by RevZero** [comment]
The official API documentation states: "...can be retrieved via the ISteamUGC API using a combination of the `m_unNumChildren` member variable of the `SteamUGCDetails_t` struct and `GetQueryUGCChildren`."

I did that and requesting the workshop mod dependencies using `GetQueryUGCChildren` works in general. I get a list containing those items. BUT: **`m_unNumChildren` from the `SteamUGCDetails_t` is always returned as 0**. I tried that for lots of different mods. 

I could just query the first 10 items from the list (as I did during testing), but the identifiers from the list seem to be kind of random. The only way to check if those are actual XCOM2 workshop items would be to request the `SteamUGCDetails_t` for each one and check the data. But that is massive overhead because the request takes pretty long to finish. And it would be kind of dirty, because `m_unNumChildren` SHOULD work.

I tried lots of stuff, but ran out of ideas.
